simulator/mips/mips_instr.cpp

```diff
@@ -458,31 +487,43 @@ void MIPSInstr::execute()
         std::ostringstream oss;
         oss << "\t [ $" << std::hex;
         if ( dst.is_mips_hi_lo())
-            oss <<  MIPSRegister::mips_hi << " = 0x" << static_cast<uint32>( v_dst >> 32) << ", $"
+            oss <<  MIPSRegister::mips_hi << " = 0x" << static_cast<uint64>( v_dst >> 32) << ", $"
```

Comment:
Is shift still valid?
I suggest to use Denis's function `get_adapted_data` here